webapp-django/crashstats/crashstats/static/crashstats/js/socorro/correlation.js

```diff
@@ -97,4 +97,169 @@ $(document).ready(function () {
             return 'MODULE: ' + correlationItem(moduleLine);
         }
     };
+
+    function hex(buffer) {
+        let hexCodes = [];
+        let view = new DataView(buffer);
+
+        for (let i = 0; i < view.byteLength; i += 4) {
+            // Using getUint32 reduces the number of iterations needed (we process 4 bytes each time).
+            let value = view.getUint32(i);
+            // toString(16) will give the hex representation of the number without padding.
+            let stringValue = value.toString(16);
+            // We use concatenation and slice for padding.
+            let padding = '00000000';
+            let paddedValue = (padding + stringValue).slice(-padding.length);
+            hexCodes.push(paddedValue);
+        }
+
+        // Join all the hex strings into one
+        return hexCodes.join('');
+    }
+
+    function sha1(str) {
+        return crypto.subtle.digest('SHA-1', new TextEncoder('utf-8').encode(str))
+        .then(hash => hex(hash));
+    }
+
+    let correlationData = {};
```

Comment:
I'm caching the results in this variable, so they don't have to be requested multiple times (e.g. if in the correlations tab the user switches to another product and then switches back).